integration-tests/test/rpc.spec.ts

```diff
@@ -233,6 +234,53 @@ describe('Basic RPC tests', () => {
 
       expect(res).to.eq(BigNumber.from(value))
     })
+
+    // https://github.com/ethereum-optimism/optimism/issues/1998
+    it('should use address(0) as the default "from" value', async () => {
+      // Deploy a contract to check msg.caller
+      const Factory__ValueContext: ContractFactory =
+        await ethers.getContractFactory('ValueContext', wallet)
+      const ValueContext: Contract = await Factory__ValueContext.deploy()
+      await ValueContext.deployTransaction.wait()
+
+      // Do the call and check msg.sender
+      const data = ValueContext.interface.encodeFunctionData('getCaller')
```

Comment:
I believe you should be able to do `await ValueContext.getCaller()` and it should handle the abi encoding/decoding for you

Reply:
I considered it at first, but decided against it because I believe that being explicit is best (especially in tests).

If I'm not mistaken in ethers, `ValueContext.getCaller()` will invoke `populateTransaction()` , which, on the other hand, [assigns a default from value](https://github.com/ethers-io/ethers.js/blob/5899c8aec0600dca42d7e24fc215037aa48917f2/packages/contracts/src.ts/index.ts#L205).